modules/dnn/src/layers/layer_norm.cpp

```diff
@@ -20,11 +21,13 @@ class LayerNormLayerImpl CV_FINAL : public LayerNormLayer
 
         // opencv attr
         hasBias = params.get<bool>("hasBias", false);
+        is1D = params.get<bool>("is1D", false);
```

Comment:
If I'm not mistaken, in case of 1D `cv::Mat` it has N rows and 1 column (`Nx1`). So with `axis = 0`  it should be correct. I think using such shortcut from MVN is enough:
https://github.com/opencv/opencv/blob/2ff16d4c45b122bef659eee4b0ee2fa970b3bfae/modules/dnn/src/layers/mvn_layer.cpp#L331-L346

Reply:
@dkurt, in 4.x there is no support for 1D. In 5.x I have added support for 1D and 0D matrices. 1D matrix of N elements may be interpreted (if you ignore Mat::dims) as 1xN 2D matrix, i.e. it's a single-row matrix, not single-column

Reply:
My point is that instead of extra `is1D` flag there is a check that let skip layer compute because output values will be all zeros.

Reply:
The 1d flag is introduced for other backends to avoid incorrect shape inference. Let me try to bypass this...

Lets say axis=-1 and the shape of scale mat is still 2d; that stands for a 1d scale and bias tensor.